core/src/main/java/io/grpc/transport/AbstractClientStream.java

```diff
@@ -45,9 +47,12 @@
 import java.util.logging.Level;
 import java.util.logging.Logger;
 
+import javax.annotation.concurrent.NotThreadSafe;
+
 /**
  * The abstract base class for {@link ClientStream} implementations.
  */
+@NotThreadSafe
```

Comment:
Why not do both? Adding `NotThreadSafe` seems fine to me. The existence of the annotation doesn't automatically imply that it's inverting the default assumption about classes ... they'll still be assumed to be not thread safe unless specified otherwise.